examples/eks-msk-opensearch/terraform/templates/fluent-bit.yaml

```diff
@@ -0,0 +1,89 @@
+# Template out the image version and service account name to use so that they can be modified from Terraform.
+image:
+  tag: ${image_version}
+
+serviceAccount:
+  name: ${service_account_name}
+
+# Tolerate all taints so that all nodes are logged.
+tolerations:
+  - key: ""
+    operator: "Exists"
+    effect: "NoExecute"
+
+# Pipeline configuration.
+config:
+  # This is a config for the FluentBit application running the pipeline.
+  # We set the flush interval, log level and where our parsers are stored.
+  # We additionally expose the HTTP service so that Terraform can ping it.
+  service: |
+    [SERVICE]
+        Daemon Off
+        Flush 1
+        Log_Level info
+        Parsers_File parsers.conf
+        HTTP_Server On
+        HTTP_Listen 0.0.0.0
+        HTTP_Port 2020
+        Health_Check On
+  # We read all of the logs for output by docker's json file logging engine.
+  # Every line appended to a file that matches the Path field will generate a record in the pipeline.
+  inputs: |
+    [INPUT]
```

Comment:
I always use the container insights daemonset as a starting point: https://github.com/aws-samples/amazon-cloudwatch-container-insights/blob/master/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/fluent-bit/fluent-bit.yaml#L65

Unless you have a good reason not to copy from this template, copy it and then modify. The tail input here should be almost exactly the same with same settings. And the [SERVICE] section is needed too. 

All of the things the template enables by default should be enabled for all EKS Fluent Bit deployments unless there is a good reason not to. Especially the prom monitoring server in [SERVICE] and the filesystem buffering options.